### ISSUE TITLE
Upcoming Puppet 3.7 / PE 3.4 x64 compatibility changes

### ISSUE THREAD

**Issue by Iristyle** [basti1302/puppet-windows-path#5]
As part of making Puppet compatible with Windows x64 / Ruby 2, we have had to make a number of changes to our included gems, and have removed `sys-admin`, `windows-pr`, `win32-api` and `windows-api`.  Other gems have been upgraded to versions that require FFI and are compatible with x64.

We don't have builds released yet, but keep an eye on https://groups.google.com/forum/#!forum/puppet-announce

I have identified a small issue with the code in this module that will require some updating to ensure it maintains compatibility with both existing Puppet releases and the upcoming releases.  I'm getting in touch with you now, so that you're not caught off guard, and so that your module is ready to go by the time 3.7 ships.

Issues:
- Defines the `SendMessageTimeout` API call using Ruby's `Win32API`  class, which won't work on x64 due to incorrect definitions for variables that have a platform specific width (such as `HANDLE`).  These API calls should use FFI definitions, so that they are properly defined on both x86 and x64.  Many examples exist in the Puppet codebase now in the `lib/puppet/util/windows` folder, such as https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/process.rb#L214

We will be making an announcement to the puppet-dev list shortly mentioning some of these upcoming changes.
https://groups.google.com/forum/#!forum/puppet-dev

Let me know if you have any questions.

Thanks!

**Comment by logicminds** [comment]
@basti1302 was this module updated to take care of these issues?

**Comment by basti1302** [comment]
No, it wasn't. I'm not using puppet anymore, and I'm not maintaining this module actively anymore. If you would like to take care of this, go ahead. I would merge it in a timely fashion and/or add you as a collaborator, if you are interested.

**Comment by basti1302** [comment]
Closing as I no longer maintain this (also stated in the README now)

**Comment by basti1302** [comment]
Reopening as there is a new maintainer soon.
